src/main/java/org/jenkinsci/plugins/ghprb/GhprbAdditionalParameterEnvironmentContributor.java

```diff
@@ -0,0 +1,102 @@
+package org.jenkinsci.plugins.ghprb;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.EnvironmentContributor;
+import hudson.model.ParameterValue;
+import hudson.model.ParametersAction;
+import hudson.model.Run;
+import hudson.model.StringParameterValue;
+import hudson.model.TaskListener;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+@Extension(optional = true)
+public class GhprbAdditionalParameterEnvironmentContributor extends EnvironmentContributor {
+
+    /*
+     * Fail the load of this extension if ParametersAction.getAllParameter() method doesn't exist
+     */
+    static {
+        try {
+            ParametersAction.class.getMethod("getAllParameters");
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void buildEnvironmentFor(@Nonnull Run r, @Nonnull EnvVars envs, @Nonnull TaskListener listener)
+            throws IOException, InterruptedException {
+        GhprbCause cause = (GhprbCause) r.getCause(GhprbCause.class);
+        ParametersAction pa = r.getAction(ParametersAction.class);
+        if (cause != null && pa != null) {
+            final String commitSha = cause.isMerged() ? "origin/pr/" + cause.getPullID() + "/merge" : cause.getCommit();
```

Comment:
You first just need to check if the var is already in envs if you keep the old core dep.